Source/WebCore/platform/graphics/coreimage/FEMorphologyCoreImageApplier.mm

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "FEMorphologyCoreImageApplier.h"
+
+#if USE(CORE_IMAGE)
+
+#import "FEMorphology.h"
+#import "Filter.h"
+#import "FilterImage.h"
+#import "Logging.h"
+#import <CoreImage/CoreImage.h>
+#import <wtf/TZoneMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_ALLOCATED_IMPL(FEMorphologyCoreImageApplier);
+
+FEMorphologyCoreImageApplier::FEMorphologyCoreImageApplier(const FEMorphology& effect)
+    : Base(effect)
+{
+    ASSERT(supportsCoreImageRendering(effect));
+}
+
+bool FEMorphologyCoreImageApplier::supportsCoreImageRendering(const FEMorphology& effect)
+{
+    return effect.radiusX() > 0 || effect.radiusY() > 0;
```

Comment:
I think we should still claim to support CI rendering, and just be pass-through (if we indeed get here).